src/rpcrawtransaction.cpp

```diff
@@ -24,110 +23,373 @@ extern std::string GetTxProject(uint256 hash, int& out_blocknumber, int& out_blo
+                    // This gives us an accruate vote shares at that time. We like to keep wallet information as accruate as possible.
+                    // Note during boosted superblocks we get unusual calculations for total network magnitude.
+                    CBlockIndex* pblockindex = mapBlockIndex[mtx.hashBlock];
+                    CBlock block;
+
+                    int nEndHeight = pblockindex->nHeight - (BLOCKS_PER_DAY*14);
+
+                    // Incase; Why throw.
+                    if (nEndHeight < 1)
+                        nEndHeight = 1;
+
+                    // Iterate back to find previous superblock
+                    while (pblockindex->nHeight > nEndHeight && pblockindex->nIsSuperBlock == 0)
+                        pblockindex = pblockindex->pprev;
+
+                    if (pblockindex->nIsSuperBlock == 1)
+                    {
+                        block.ReadFromDisk(pblockindex);
+
+                        std::string sHashBoinc = block.vtx[0].hashBoinc;
+
+                        MiningCPID vbb = DeserializeBoincBlock(sHashBoinc, block.nVersion);
+
+                        std::string sUnpackedSuperblock = UnpackBinarySuperblock(vbb.superblock);
+                        std::string sMagnitudeContract = ExtractXML(sUnpackedSuperblock, "<MAGNITUDES>", "</MAGNITUDES>");
+
+                        // Since Superblockavg function gives avg for mags yes but total cpids we cannot use this function
+                        // We need the rows_above_zero for Total Network Magnitude calculation with Money Supply Factor.
+                        std::vector<std::string> vMagnitudeContract = split(sMagnitudeContract, ";");
+                        int nRowsWithMag = 0;
+                        double dTotalMag = 0;
+
+                        for (auto const& sMag : vMagnitudeContract)
+                        {
+                            const std::vector<std::string>& vFields = split(sMag, ",");
+
+                            if (vFields.size() < 2)
+                                continue;
+
+                            const std::string& sCPID = vFields[0];
+                            double dMAG = std::stoi(vFields[1].c_str());
+
+                            if (sCPID.length() > 10)
+                            {
+                                nRowsWithMag++;
+                                dTotalMag += dMAG;
+                            }
+                        }
+
+                        double dOutAverage = dTotalMag / ((double)nRowsWithMag + .01);
+                        double dTotalNetworkMagnitude = (double)nRowsWithMag * dOutAverage;
+                        double dMoneySupply = DoubleFromAmount(pblockindex->nMoneySupply);
+                        double dMoneySupplyFactor = (dMoneySupply/dTotalNetworkMagnitude + .01);
+
+                        dVoteMagnitude = cdbl(sVoteMagnitude,2);
+                        dVoteBalance = cdbl(sVoteBalance,2);
```

Comment:
RoundFromString

Reply:
yea i'll add that to my list that was existing code I used for that little part